libpod/boltdb_state_internal.go

```diff
@@ -407,11 +407,16 @@ func (s *BoltState) addContainer(ctr *Container, pod *Pod) error {
 				return errors.Wrapf(ErrNoSuchCtr, "container %s depends on container %s, but it does not exist in the DB", ctr.ID(), dependsCtr)
 			}
 
-			// If we're part of a pod, make sure the dependency is part of the same pod
+			depCtrPod := depCtrBkt.Get(podIDKey)
```

Comment:
Not part of this review, but the pod variable will be nil if we're in a pod, and then depCtrPod will be nil if we're in that same pod?  That seems counter-intuitive.

Reply:
I think you've got this reversed - pod will be non-nil if we're in a pod, and pod is not nil depCtrPod must also be non-nil (we must be part of a pod if our dependency is part of a pod).

Now that I think of it, though, this isn't a strong enough restriction, we should actually check that the ID of the pods match too. Let me fix that.

Reply:
TY, the code is a lot cleaner with this new approach.